lib/phoenix/tracker.ex

```diff
@@ -62,13 +62,34 @@ defmodule Phoenix.Tracker do
   An optional `handle_info/2` callback may also be invoked to handle
   application specific messages within your tracker.
 
-  ## Special Considerations
+  ## Stability and Performance Considerations
 
   Operations within `handle_diff/2` happen *in the tracker server's context*.
   Therefore, blocking operations should be avoided when possible, and offloaded
   to a supervised task when required. Also, a crash in the `handle_diff/2` will
   crash the tracker server, so operations that may crash the server should be
   offloaded with a `Task.Supervisor` spawned process.
+
+  ## Application Shutdown
+
+  A tracker does not automatically replicate its state across the cluster as it
+  shuts down. This means that your supervision tree shuts down normally - as it
+  does when you call `System.stop()` or the BEAM receives a `SIGTERM` - any
+  presences that the local tracker instance has will continue to be seen as
+  present by other trackers in the cluster until the `:down_period` for the
+  instance has passed.
+
+  If you want a normal shutdown to immediately cause other nodes to see that
+  tracker's presences as leaving, pass `permdown_on_shutdown: true`. On the
+  other hand, if you are using `Phoenix.Presence` for clients which will
+  immediately attempt to connect to a new node, it may be preferable to use
+  `permdown_on_shutdown: false`, allowing the disconnected clients time to
+  reconnect before removing their old presences, to avoid overwhelming clients
+  with notifications that many users left and immediately rejoined.
+
+  If the application crashes or is halted non-gracefully (for instance, with a
+  `SIGKILL` or a `Ctrl+C` in `iex`), other nodes will still have to wait the
+  `:down_period` to notice that the tracker's presences are gone.
```

Comment:
```suggestion
  When a tracker shuts down, the other nodes do not assume it has been gone
  for good. After all, in a distributed system, it is impossible to know if something
  is just temporarily unavailable or if it has crashed.

  For this reason, when you call `System.stop()` or the Erlang VM receives a
  `SIGTERM` - any presences that the local tracker instance has will continue to
  be seen as present by other trackers in the cluster until the `:down_period`
  for the instance has passed.

  If you want a normal shutdown to immediately cause other nodes to see that
  tracker's presences as leaving, pass `permdown_on_shutdown: true`. On the
  other hand, if you are using `Phoenix.Presence` for clients which will
  immediately attempt to connect to a new node, it may be preferable to use
  `permdown_on_shutdown: false`, allowing the disconnected clients time to
  reconnect before removing their old presences, to avoid overwhelming clients
  with notifications that many users left and immediately rejoined.

  If the application crashes or is halted non-gracefully (for instance, with a
  `SIGKILL` or a `Ctrl+C` in `iex`), other nodes will still have to wait the
  `:down_period` to notice that the tracker's presences are gone.
```